static/app/components/idBadge/teamBadge/index.tsx

```diff
@@ -8,6 +9,14 @@ import Badge, {BadgeProps} from './badge';
 function TeamBadge(props: BadgeProps) {
   const [team, setTeam] = React.useState<Team>(props.team);
 
+  React.useEffect(() => {
```

Comment:
@scttcper when we mutate a team, do we only override a single property like 
```
team.property = newproperty
```
If we instead treat this in a more immutable way, we should have this component update without the need for the effect or isEqual. So something like this:
```
newTeam = {...team, property: newproperty}
```
if we do the former, then we need to be careful, as this effect wont run either because hook dependencies do not do shallow comparison hence [props.team] points to the same old obj and we should instead set [team.name, ...] or any props that we want to actually observe.

I'm sorry for the breaking change, I'll be more careful next time!